cirq/algorithms/qaoa/QAOA.py

```diff
@@ -0,0 +1,353 @@
+# Cirq Quantum Approximate Optimization
+#
+# Fabio Sanches - QC Ware
+
+# imports
+import numpy as np
+import pyswarm
+
+# import for cirq with new functionality - change path here
+# import sys
+# sys.path.insert(0,'PATH')
+# cirq has to come after this import
+
+from scipy.optimize import minimize
+from cirq.google import XmonSimulator, XmonMeasurementGate
+from cirq.circuits import Circuit
+from cirq.ops import RotXGate, RotYGate, MeasurementGate, H, Pauli, PauliString
+from cirq.devices import GridQubit
+# from cirq.circuits import exponentiate, expectation_value
+from exponentiate import exponentiate_qubit_operator
+from expectation_value import expectation_value, expectation_from_sampling
+from bayes_opt import BayesianOptimization
+
+
+# QAOA class
+class QAOA:
+    """
+    A class for the quantum approximate optimization algorithm
+        based on https://arxiv.org/pdf/1411.4028.pdf
+
+    """
+
+    def __init__(self, n_qubits, objective_function, mixing_operator='X'):
+        """
+        Initializes the QAOA object
+
+        :param n_qubits: number of qubits used in the circuit
```

Comment:
Use the argument declaration style from the style guide: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#383-functions-and-methods (i.e. an `Args:` block)